spock-spring/src/main/java/org/spockframework/spring/SpringInterceptor.java

```diff
@@ -34,6 +34,7 @@ public SpringInterceptor(SpringTestContextManager manager) {
   @Override
   public void interceptSetupSpecMethod(IMethodInvocation invocation) throws Throwable {
     manager.beforeTestClass();
+    manager.prepareTestInstance(invocation.getInstance());
```

Comment:
I think we should actually guard this with a check for `isAnnotationPresent(EnableSharedInjection.class)`, so that it behaves as before if the annotation is not present.

One possible way to verify that is to add a custom `CustomTestExecutionListener` via 
```java
@TestExecutionListeners(
  value = { CustomTestExecutionListener.class }, 
  mergeMode = MergeMode.MERGE_WITH_DEFAULTS)
```

and check that `prepareTestInstance` is not called for the shared instance, the easiest way would be with a simple counter, as we would expect two calls for a spec with a single feature with `@EnableSharedInjection` and one call without.

Reply:
This is a very good point. Thanks for the pointer about testing, I would have struggled to come up with a strategy for that without it. It might take until the weekend before I will be able to get to it, though.

> Sorry that I still have one issue, but other than that it looks good.

You should have promised that it's the last one - then I would at least something to laugh about when it won't be. ;)

Reply:
This has been addressed. The test I added was failing before applying the prod code change. Please let me know if you know about a way to simplify or improve the test in any way.